Mathlib/CategoryTheory/Limits/IndYoneda.lean

```diff
@@ -0,0 +1,192 @@
+
+/-- Hom is cocontinuous: homomorphisms from a colimit is the limit over yoneda of the diagram. -/
+noncomputable def colimitHomIsoLimitYoneda
+    [HasLimitsOfShape Iᵒᵖ (Type u₂)] (A : C) :
+    (colimit F ⟶ A) ≅ limit (F.op ⋙ yoneda.obj A) :=
+  ((coyonedaOpColimitIsoLimitCoyoneda F).app A).trans <| limitObjIsoLimitCompEvaluation _ _
+
+lemma colimitHomIsoLimitYoneda_comp_π [HasLimitsOfShape Iᵒᵖ (Type u₂)] (A : C) (i : I) :
+    (colimitHomIsoLimitYoneda F A).hom ≫ limit.π (F.op ⋙ yoneda.obj A) ⟨i⟩
+      = (coyoneda.map (colimit.ι F i).op).app A := by
+  simp only [colimitHomIsoLimitYoneda, Iso.trans_hom, Iso.app_hom, Category.assoc]
+  erw [limitObjIsoLimitCompEvaluation_hom_π]
+  change ((coyonedaOpColimitIsoLimitCoyoneda F).hom ≫ _).app A = _
+  rw [coyonedaOpColimitIsoLimitCoyoneda_comp_π]
+
+end HomCocontinuousCovariant
+
+section HomCocontinuousContravariant
+
+variable (F : Iᵒᵖ ⥤ C) [HasColimit F]
+
+/-- Variant of `coyonedaOoColimitIsoLimitCoyoneda` for contravariant `F`. -/
+noncomputable def coyonedaOpColimitIsoLimitCoyoneda' :
+    coyoneda.obj (op <| colimit F) ≅ limit (F.rightOp ⋙ coyoneda) :=
+  coyoneda.mapIso (limitRightOpIsoOpColimit F).symm ≪≫ preservesLimitIso coyoneda F.rightOp
+
+@[simp]
+lemma coyonedaOpColimitIsoLimitCoyoneda'_comp_π (i : I) :
+    (coyonedaOpColimitIsoLimitCoyoneda' F).hom ≫ (limit.π (F.rightOp ⋙ coyoneda) i)
+      = (coyoneda.map (colimit.ι F ⟨i⟩).op) := by
+  simp only [coyonedaOpColimitIsoLimitCoyoneda', Functor.mapIso_symm, Iso.trans_hom, Iso.symm_hom,
+    Functor.mapIso_inv, Category.assoc, preservesLimitsIso_hom_π, ← Functor.map_comp,
+    limitRightOpIsoOpColimit_inv_comp_π]
+
+/-- Variant of `colimitHomIsoLimitYoneda` for contravariant `F`. -/
+noncomputable def colimitHomIsoLimitYoneda' [HasLimitsOfShape I (Type u₂)] (A : C) :
+    (colimit F ⟶ A) ≅ limit (F.rightOp ⋙ yoneda.obj A) :=
+  (coyonedaOpColimitIsoLimitCoyoneda' F).app A ≪≫ limitObjIsoLimitCompEvaluation _ _
+
+@[simp]
+lemma colimitHomIsoLimitYoneda'_comp_π [HasLimitsOfShape I (Type u₂)] (A : C) (i : I) :
+    (colimitHomIsoLimitYoneda' F A).hom ≫ limit.π (F.rightOp ⋙ yoneda.obj A) i
+      = (coyoneda.map (colimit.ι F ⟨i⟩).op).app A := by
+  simp only [yoneda_obj_obj, colimitHomIsoLimitYoneda', Iso.trans_hom,
+    Iso.app_hom, Category.assoc]
+  erw [limitObjIsoLimitCompEvaluation_hom_π]
+  change ((coyonedaOpColimitIsoLimitCoyoneda' F).hom ≫ _).app A = _
+  rw [coyonedaOpColimitIsoLimitCoyoneda'_comp_π]
+
+end HomCocontinuousContravariant
+
+section ProCoyonedaContravariant
+
+variable (D : Iᵒᵖ ⥤ C) (F : C ⥤ Type u₂)
+variable [HasColimit (D.rightOp ⋙ coyoneda)] [HasLimitsOfShape Iᵒᵖ (Type (max u₁ u₂))]
+
+/-- Pro-Coyoneda lemma: homorphisms from colimit of coyoneda of diagram `D` to `F` is limit
+of `F` evaluated at `D`. -/
+noncomputable def colimitCoyonedaHomIsoLimit :
+    (colimit (D.rightOp ⋙ coyoneda) ⟶ F) ≅ limit (D ⋙ F ⋙ uliftFunctor.{u₁}) :=
+  (colimitHomIsoLimitYoneda _ F).trans
```

Comment:
Do you know about the notation for `Iso.trans`? (Similarly below)

Reply:
Yes, but I always forget (or maybe never knew) how to type it so I only use it when something to copy paste it from is close. I fixed it in the latest commit.